var/spack/repos/builtin/packages/fenics-dolfinx/package.py

```diff
@@ -59,15 +63,16 @@ class FenicsDolfinx(CMakePackage):
     def cmake_args(self):
         args = [
             "-DDOLFINX_SKIP_BUILD_TESTS=True",
-            "-DDOLFINX_ENABLE_KAHIP=%s" % (
-                'ON' if "+kahip" in self.spec else 'OFF'),
             "-DDOLFINX_ENABLE_PARMETIS=%s" % (
                 'ON' if "+parmetis" in self.spec else 'OFF'),
+            "-DDOLFINX_ENABLE_KAHIP=%s" % (
+                'ON' if "+kahip" in self.spec else 'OFF'),
```

Comment:
Have you considered using any of the `CMakePackage` helper methods like `define_from_variant()`?  

For more, see https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake.